sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -115,6 +115,11 @@ final class CastExprMeta[INPUT <: CastBase](
         YearParseUtil.tagParseStringAsDate(conf, this)
       case (_: StringType, _: DateType) =>
         YearParseUtil.tagParseStringAsDate(conf, this)
+      case (_: StringType, dt:DecimalType) =>
+        if (dt.scale < 0 && !ShimLoader.getSparkShims.isNegativeDecimalScaleSupportEnabled) {
+          willNotWorkOnGpu("Rapids doesn't support negative decimal scale for this version of " +
```

Comment:
```suggestion
          willNotWorkOnGpu("RAPIDS doesn't support negative decimal scale for this version of " +
```

Reply:
"RAPIDS doesn't support casting string to decimal for negative scale decimal in this version of Spark because of SPARK-....